### PR TITLE
PLD: common upload code for create and patch, no params for patch, wa…

### PR DESCRIPTION
**Pull request by pld**
…s unused

### CODE REVIEW

File: src/milia/api/dataset.cljc [hunk 1]
```diff
@@ -18,34 +18,32 @@
     (parse-http :get url)))
 
 #?(:clj
+   (defn- maybe-request-file
```
Comment (Santei) [inline src/milia/api/dataset.cljc]: Mmmh,function name could be something like `get-file`.

Comment (pld) [inline src/milia/api/dataset.cljc]: It's not necessarily getting a file though, what about: `send-file-or-params`?

File: tests/clj/milia/api/dataset_test.clj [hunk 2]
```diff
@@ -229,30 +233,31 @@
...
+                 (patch :dataset-id file) => :response
+                 (provided
+                  (make-url "forms" :dataset-id) => url
+                  (multipart-options :uploaded-file "xls_file")
+                  => multipart-options-map
+                  (parse-http :patch
+                              url
+                              :http-options multipart-options-map
+                              :suppress-4xx-exceptions? true)
+                  => :response))))
```
Comment (Santei) [inline tests/clj/milia/api/dataset_test.clj]: Looks better this way `:suppress-4xx-exceptions? true) => :response))`

**Comment by Santei** [comment]
Looks good.

File: src/milia/api/dataset.cljc (hunk 1 above)
Comment (Santei) [inline src/milia/api/dataset.cljc]: That's better.

**Pull request merged**